From 51a51d46f666f1cf70307237f1a6b176a09849db Mon Sep 17 00:00:00 2001
From: Lyon Wang <lyon.wang@nxp.com>
Date: Fri, 9 Jun 2017 16:53:11 +0800
Subject: [PATCH] [gstaacparse] Fix aacparse adif profile fetch error

- Fix aacparse src pad caps wrong profile for ADIF file

Upstream-Status: Submitted
Bugzilla ticket:
https://bugzilla.gnome.org/show_bug.cgi?id=783583

Signed-off-by: Lyon Wang <lyon.wang@nxp.com>
---
 gst/audioparsers/gstaacparse.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/gst/audioparsers/gstaacparse.c b/gst/audioparsers/gstaacparse.c
index 09a53cc..98424d2 100644
--- a/gst/audioparsers/gstaacparse.c
+++ b/gst/audioparsers/gstaacparse.c
@@ -1013,13 +1013,13 @@ gst_aac_parse_detect_stream (GstAacParse * aacparse,
 
       GST_INFO ("ADIF buffer fullness: %d", fullness);
 #endif
-      aacparse->object_type = ((adif[6 + skip_size] & 0x01) << 1) |
-          ((adif[7 + skip_size] & 0x80) >> 7);
+      aacparse->object_type = (((adif[6 + skip_size] & 0x01) << 1) |
+          ((adif[7 + skip_size] & 0x80) >> 7)) + 1;
       sr_idx = (adif[7 + skip_size] & 0x78) >> 3;
     }
     /* VBR */
     else {
-      aacparse->object_type = (adif[4 + skip_size] & 0x18) >> 3;
+      aacparse->object_type = ((adif[4 + skip_size] & 0x18) >> 3) + 1;
       sr_idx = ((adif[4 + skip_size] & 0x07) << 1) |
           ((adif[5 + skip_size] & 0x80) >> 7);
     }
-- 
1.9.1

